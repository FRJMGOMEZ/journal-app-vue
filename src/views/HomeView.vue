<template>
  <div class="home">
    <img alt="Vue logo" src="../assets/logo.png">
    <h1>Hola mundo</h1>
    <button class="btn btn-primary"> primer</button>
    <button class="btn btn-secondary"> segundo</button>
    <button class="btn btn-warning">tercero</button>
  </div>
</template>

<script>
// @ is an alias to /src


export default {
  name: 'HomeView',
  components: {
    
  }
}
</script>
