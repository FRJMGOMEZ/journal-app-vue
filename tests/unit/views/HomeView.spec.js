
import HomeView from '@/views/HomeView.vue';
import { shallowMount } from '@vue/test-utils';

describe('HomeView component', () => {

    test('debe de hacer match con snapshot', () => {
        const wrapper = shallowMount(HomeView);
        expect(wrapper.html()).toMatchSnapshot();
    })

    test('Hacer click en un botÃ³n, debe de redireccionar a no-entry',()=>{
       const mockRouter = {
        push:jest.fn()
       };
       const wrapper = shallowMount(HomeView,{global:{mocks:{$router:mockRouter}}});
       wrapper.find('button').trigger('click');
       expect(mockRouter.push).toHaveBeenCalled();
       expect(mockRouter.push).toHaveBeenCalledWith({name:'no-entry'});
    })
})
