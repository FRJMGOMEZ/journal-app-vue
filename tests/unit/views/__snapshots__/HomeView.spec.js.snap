// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView component debe de hacer match con snapshot 1`] = `
<div class="home"><img alt="Vue logo" src="">
  <h1>Hola mundo</h1><button class="btn btn-primary"> Daybook</button><button class="btn btn-secondary"> segundo</button><button class="btn btn-warning">tercero</button>
</div>
`;
